app/controllers/alert_requests_controller.rb

```diff
@@ -0,0 +1,114 @@
+class AlertRequestsController < ApplicationController
+  layout 'alert_request'
+  before_action :find_partner
+
+  def new
+    @alert_request = AlertRequest.new(
+      partner_id: @partner_id, 
+      tracking_source: @source,
+      tracking_id: @tracking,
+      phone_type: 'mobile',
+    )
+
+    # do we need this?
+    # if @alert_request.partner.primary?
+    #   @alert_request.opt_in_email = true
+    # else
+    #   if @alert_request.partner.rtv_email_opt_in?
+    #     @alert_request.opt_in_email = true
+    #   end
+    #   if @alert_request.partner.partner_email_opt_in?
+    #     @alert_request.partner_opt_in_email = true
+    #   end
+    # end
+  
+    set_up_locale
+    @registrant = OpenStruct.new
+    @question_1 = @alert_request.question_1
+    @question_2 = @alert_request.question_2
+  end
+
+  def create
+    @alert_request = AlertRequest.create(
+      alert_request_params.merge(
+        uid: SecureRandom.hex(20),
```

Comment:
Lets use a generate_uid method in registrant and abr and put it on the AlertRequest model (with before_create: generate_uid)